Fix:
y.tab.c:1310:16: error: implicit declaration of function 'yylex' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
      yychar = YYLEX;
               ^
y.tab.c:1586:7: error: implicit declaration of function 'yyerror' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
      yyerror (YY_("syntax error"));
      ^
y.tab.c:1732:3: error: implicit declaration of function 'yyerror' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
  yyerror (YY_("memory exhausted"));
  ^
--- src/tclx/src/tclxgdat.y.orig	2018-06-29 04:19:31.000000000 -0500
+++ src/tclx/src/tclxgdat.y	2022-05-16 20:59:32.000000000 -0500
@@ -58,6 +58,9 @@
 #define DAYLIGHT 1
 #define STANDARD 2
 #define MAYBE    3
+
+int yylex(void);
+void yyerror(const char *);
 %}
 
 %%
@@ -229,6 +232,7 @@
 static char *lptr;
 
 //static
+int
 yylex()
 {
 #ifndef YYSTYPE
@@ -580,7 +584,7 @@
  * Error message are not used, so discard with dummy function.
  */
 
-int
+void
 yyerror(msg)
     const char *msg;
 {
